cdap-api/src/main/java/co/cask/cdap/api/dataset/DatasetDefinition.java

```diff
@@ -71,24 +74,28 @@
    * Provides dataset admin to be used to perform administrative operations on the dataset instance defined by passed
    * {@link DatasetSpecification}.
    *
-   * @param spec specification of the dataset instance.
+   *
+   * @param datasetContext context for the dataset
    * @param classLoader classloader to use when executing admin operations
+   * @param spec specification of the dataset instance.
    * @return dataset admin to perform administrative operations
    * @throws IOException
    */
-  A getAdmin(DatasetSpecification spec, ClassLoader classLoader) throws IOException;
+  A getAdmin(DatasetContext datasetContext, ClassLoader classLoader, DatasetSpecification spec) throws IOException;
```

Comment:
What will happen if I create dataset in namespace A and in this method provide the context with namespace B?
With this change DatasetSpecification no longer identifies dataset instance, but rather dataset instance is identified by a pair: context and spec. How do we make sure these two are "in sync", meaning that we don't provide wrong context with specific spec and the other way around?